modules/components/src/Aggs/DatesAgg.js

```diff
@@ -0,0 +1,293 @@
+    left: 0%;
+    bottom: 100%;
+  `,
+  TOP_RIGHT: css`
+    position: absolute;
+    right: 0%;
+    bottom: 100%;
+  `,
+};
+
+const Arrow = ({ style }) => (
+  <svg
+    style={style}
+    className="DayPickerNavigation_svg__horizontal DayPickerNavigation_svg__horizontal_1"
+    viewBox="0 0 1000 1000"
+  >
+    <path d="M694.4 242.4l249.1 249.1c11 11 11 21 0 32L694.4 772.7c-5 5-10 7-16 7s-11-2-16-7c-11-11-11-21 0-32l210.1-210.1H67.1c-13 0-23-10-23-23s10-23 23-23h805.4L662.4 274.5c-21-21.1 11-53.1 32-32.1z" />
+  </svg>
+);
+
+class DatesAgg extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      localRange: { ...this.getInitialRange(props) }, // the moment objects passed into DayPickerRangeController
+      inputRangeValues: {
+        // the strings to render in the inputs
+        startDate: null,
+        endDate: null,
+      },
+      focusedInput: null, // must be set with this.setInputFocus to bind with dom state
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.setState({
+      localRange: { ...this.getInitialRange(nextProps) },
+    });
+  }
+
+  getInitialRange = ({
+    buckets = [],
+    startDateFromSqon = () => null,
+    endDateFromSqon = () => null,
+  }) => {
+    const bucketWithMoment = buckets.map(({ key_as_string, ...rest }) => ({
+      ...rest,
+      key_as_string,
+      moment: bucketDateToMoment(key_as_string),
+    }));
+    const minBucket = minBy(bucketWithMoment, ({ moment }) => moment.valueOf());
+    const maxBucket = maxBy(bucketWithMoment, ({ moment }) => moment.valueOf());
+
+    const startFromSqon = startDateFromSqon({
+      getDateFromSqon: this.getDateFromSqon(START_DATE_INPUT),
+    });
+    const endFromSqon = endDateFromSqon({
+      getDateFromSqon: this.getDateFromSqon(END_DATE_INPUT),
+    });
+    return {
+      startDate:
+        (startFromSqon && bucketDateToMoment(startFromSqon)) ||
+        minBucket?.moment,
+      endDate:
+        (startFromSqon && bucketDateToMoment(endFromSqon)) || maxBucket?.moment,
+    };
+  };
+
+  setInputFocus = focusedInput => {
+    this.setState({ focusedInput }, () => {
+      if (focusedInput === START_DATE_INPUT) {
+        this.startDateInput.focus();
+      } else if (focusedInput === END_DATE_INPUT) {
+        this.endDateInput.focus();
+      } else {
+        this.startDateInput.blur();
+        this.endDateInput.blur();
+      }
+    });
+  };
+
+  getDateFromSqon = dateToGet => sqon => {
+    const { field } = this.props;
+    return sqon?.content
+      ?.filter(({ content: { field: sqonField } }) => {
+        return sqonField === field;
+      })
+      ?.find(({ op }) => op === (dateToGet === START_DATE_INPUT ? '>=' : '<='))
+      ?.content.value;
+  };
+
+  handleInputValueChange = ({ value, input }) => {
+    const newMoment = inputDateToMoment(value);
+    const isValidInputDateString =
+      newMoment.isValid() &&
+      sumBy(value.split('/'), str => str.length === 2) === 3;
```

Comment:
took me a second to realize what this line was doing. Might be a bit clearer to use a regex or something. I think it's checking for `value.match(/^\d\d\/\d\d\/\d\d$/)`?